### ISSUE TITLE
Build AdvrCommandMessage.cpp fails

### ISSUE THREAD

**Issue by DaniloCaporale** [ADVRHumanoids/XCM#58]
Got this error on brand new superbuild:

danilo@walkmanbox:~/projects/advr-superbuild/build/external/XCM (master)$ make install
[  0%] Built target std_msgs_generate_messages_cpp
[  0%] Built target _XCM_generate_messages_check_deps_cmd_service
[  0%] Built target _XCM_generate_messages_check_deps_CommandAdvr
[  0%] Built target _XCM_generate_messages_check_deps_advr_controller_joint_names
[  0%] Built target _XCM_generate_messages_check_deps_JointStateAdvr
[  0%] Built target _XCM_generate_messages_check_deps_status_service
[  5%] Built target XCM_generate_messages_cpp
[  5%] Building CXX object CMakeFiles/AdvrCommandMessage.dir/src/MessageInterfaces/AdvrCommandMessage.cpp.o
In file included from /home/danilo/projects/advr-superbuild/build/install/include/SharedLibraryFactory.h:16:0,
                 from /home/danilo/projects/advr-superbuild/build/install/include/SharedLibraryClassFactory.h:13,
                 from /home/danilo/projects/advr-superbuild/build/install/include/SharedLibraryClass.h:13,
                 from /home/danilo/projects/advr-superbuild/build/install/include/XBotInterface/RobotInterface.h:26,
                 from /home/danilo/projects/advr-superbuild/external/XCM/include/XCM/MessageInterfaces/AdvrCommandMessage.h:26,
                 from /home/danilo/projects/advr-superbuild/external/XCM/src/MessageInterfaces/AdvrCommandMessage.cpp:20:
/home/danilo/projects/advr-superbuild/external/XCM/src/MessageInterfaces/AdvrCommandMessage.cpp: In function ‘void* advrcommandmessage_control_message_create()’:
/home/danilo/projects/advr-superbuild/external/XCM/src/MessageInterfaces/AdvrCommandMessage.cpp:24:74: error: invalid new-expression of abstract class type ‘XBot::CommandAdvr’
 SHLIBPP_DEFINE_SHARED_SUBCLASS(advrcommandmessage_control_message, XBot::CommandAdvr, XBot::GenericControlMessage);
                                                                          ^
In file included from /home/danilo/projects/advr-superbuild/external/XCM/src/MessageInterfaces/AdvrCommandMessage.cpp:20:0:
/home/danilo/projects/advr-superbuild/external/XCM/include/XCM/MessageInterfaces/AdvrCommandMessage.h:31:11: note:   because the following virtual functions are pure within ‘XBot::CommandAdvr’:
     class CommandAdvr : public GenericControlMessage {
           ^
In file included from /home/danilo/projects/advr-superbuild/external/XCM/include/XCM/MessageInterfaces/AdvrCommandMessage.h:25:0,
                 from /home/danilo/projects/advr-superbuild/external/XCM/src/MessageInterfaces/AdvrCommandMessage.cpp:20:
/home/danilo/projects/advr-superbuild/build/install/include/RobotInterfaceROS/GenericControlMessage.h:48:22: note: 	virtual int& XBot::GenericControlMessage::seq_id()
         virtual int& seq_id() = 0;
                      ^
CMakeFiles/AdvrCommandMessage.dir/build.make:62: recipe for target 'CMakeFiles/AdvrCommandMessage.dir/src/MessageInterfaces/AdvrCommandMessage.cpp.o' failed
make[2]: *** [CMakeFiles/AdvrCommandMessage.dir/src/MessageInterfaces/AdvrCommandMessage.cpp.o] Error 1
CMakeFiles/Makefile2:284: recipe for target 'CMakeFiles/AdvrCommandMessage.dir/all' failed
make[1]: *** [CMakeFiles/AdvrCommandMessage.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2

Tried many workarounds with @alessandrosettimi, can't make it work. Any ideas? Could it be related to missing iit-walkman-ros-pkg missing? @liesrock

**Comment by alessandrosettimi** [comment]
As I can see, doing the following automatically activate XCM compilation:

`mkdir build`
`cd build`
`ccmake ..`
`ADVR_CORE ON`
`c`

**Comment by liesrock** [comment]
Check TRAVIS : https://travis-ci.org/ADVRHumanoids/advr-superbuild

There is only a target XCM inside XBotCore, not a repo. Try to remove the build.

**Comment by liesrock** [comment]
Check that RobotInterfaceROS and all the other repos listed here are on the latest master:

```
iit-walkman-ros-pkg/  -->  (master)
ADVR_ROS/  -->  (master)
DebrisRemovalTask/  -->  (master)
gazebo_magnetic_attach_plugin/  -->  (master)
GazeboXBotPlugin/  -->  (master)
ManipulationPlugin/  -->  (master)
MiscellaneousPlugins/  -->  (master)
ModelInterfaceRBDL/  -->  (master)
OpenSoT-lite/  -->  (v2.0)
rbdl/  -->  (master)
RobotInterfaceDUMMY/  -->  (master)
RobotInterfaceROS/  -->  (master)
RobotInterfaceXBotRT/  -->  (master)
sharedlibpp/  -->  (master)
srdfdom_advr/  -->  (master)
trajectory_utils/  -->  (master)
XBotCore/  -->  (master)
XBotCoreModel/  -->  (master)
XBotDummy/  -->  (master)
*XBotGUI/  -->  (master) 
XBotInterface/  -->  (master)
YCM/  -->  ((detached from 538bec0))
ADVR_shared/  -->  (master) 
```

**Comment by liesrock** [comment]
Very strange, TRAVIS does exaclty the same! Give me updates, I will help on this.

**Comment by DaniloCaporale** [comment]
Works on my pc now. These my options

-- Cache values
ADVR-CORE:BOOL=ON
ADVR_ROS_DIR:PATH=ADVR_ROS_DIR-NOTFOUND
ADVR_shared_DIR:PATH=ADVR_shared_DIR-NOTFOUND
BOOST_THREAD_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libpthread.so
BUILD_TESTING:BOOL=ON
CENTAURO_EU:BOOL=OFF
CMAKE_BUILD_TYPE:STRING=RelWithDebInfo
CMAKE_INSTALL_PREFIX:PATH=/usr/local
COGIMON_EU:BOOL=OFF
COMAN:BOOL=OFF
DEPRECATED_PACKAGES:BOOL=OFF
GazeboXBotPlugin_DIR:PATH=GazeboXBotPlugin_DIR-NOTFOUND
KUKA:BOOL=OFF
KUKA4LWR:BOOL=OFF
KUKAIIWA:BOOL=OFF
ManipulationExample_DIR:PATH=ManipulationExample_DIR-NOTFOUND
ManipulationPlugin_DIR:PATH=ManipulationPlugin_DIR-NOTFOUND
MiscellaneousPlugins_DIR:PATH=MiscellaneousPlugins_DIR-NOTFOUND
ModelInterfaceRBDL_DIR:PATH=ModelInterfaceRBDL_DIR-NOTFOUND
OpenSoT-lite_DIR:PATH=OpenSoT-lite_DIR-NOTFOUND
PHOLUS:BOOL=OFF
PLAYGROUND:BOOL=OFF
RobotInterfaceDUMMY_DIR:PATH=RobotInterfaceDUMMY_DIR-NOTFOUND
RobotInterfaceROS_DIR:PATH=RobotInterfaceROS_DIR-NOTFOUND
RobotInterfaceXBotRT_DIR:PATH=RobotInterfaceXBotRT_DIR-NOTFOUND
SDFormat_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/sdformat
SUPERBUILD_ADVR_ROS:BOOL=ON
SUPERBUILD_ADVR_shared:BOOL=ON
SUPERBUILD_DebrisRemovalTask:BOOL=OFF
SUPERBUILD_DoorOpening:BOOL=OFF
SUPERBUILD_EtherCAT_tools:BOOL=OFF
SUPERBUILD_GazeboXBotPlugin:BOOL=ON
SUPERBUILD_GazeboYARPPlugins:BOOL=OFF
SUPERBUILD_Head2Hand_Calib:BOOL=OFF
SUPERBUILD_Hose_Task:BOOL=OFF
SUPERBUILD_Juan_FBcontrollers:BOOL=OFF
SUPERBUILD_ManipulationExample:BOOL=ON
SUPERBUILD_ManipulationPlugin:BOOL=ON
SUPERBUILD_MiscellaneousPlugins:BOOL=ON
SUPERBUILD_ModelInterfaceIDYNUTILS:BOOL=OFF
SUPERBUILD_ModelInterfaceRBDL:BOOL=ON
SUPERBUILD_OpenSoT-lite:BOOL=OFF
SUPERBUILD_OrocosBFL:BOOL=OFF
SUPERBUILD_Playback_mat_with_DK:BOOL=OFF
SUPERBUILD_ReactiveWalk:BOOL=OFF
SUPERBUILD_RobotInterfaceDUMMY:BOOL=ON
SUPERBUILD_RobotInterfaceROS:BOOL=ON
SUPERBUILD_RobotInterfaceXBotRT:BOOL=ON
SUPERBUILD_RobotInterfaceYARP:BOOL=OFF
SUPERBUILD_SensorsPublisher:BOOL=OFF
SUPERBUILD_TestProjectGiuseppe:BOOL=OFF
SUPERBUILD_WALKMAN_shared:BOOL=ON
SUPERBUILD_WebGUIDistr:BOOL=ON
SUPERBUILD_XBotCore:BOOL=ON
SUPERBUILD_XBotCoreModel:BOOL=ON
SUPERBUILD_XBotDummy:BOOL=ON
SUPERBUILD_XBotEcat:BOOL=OFF
SUPERBUILD_XBotEthernet:BOOL=OFF
SUPERBUILD_XBotGUI:BOOL=OFF
SUPERBUILD_XBotInterface:BOOL=ON
SUPERBUILD_XBotYARP:BOOL=OFF
SUPERBUILD_advr_humanoids_common_utils:BOOL=OFF
SUPERBUILD_arms_ik:BOOL=OFF
SUPERBUILD_chengxu_walking:BOOL=OFF
SUPERBUILD_contact_planning:BOOL=OFF
SUPERBUILD_custom_sensor_msgs:BOOL=OFF
SUPERBUILD_drc_shared:BOOL=OFF
SUPERBUILD_gaze_control:BOOL=OFF
SUPERBUILD_gazebo_magnetic_attach_plugin:BOOL=ON
SUPERBUILD_iit-bigman-ros-pkg:BOOL=OFF
SUPERBUILD_iit-walkman-ros-pkg:BOOL=ON
SUPERBUILD_lazarus:BOOL=OFF
SUPERBUILD_locoman_utils:BOOL=OFF
SUPERBUILD_locomotion:BOOL=OFF
SUPERBUILD_multi_contacts:BOOL=OFF
SUPERBUILD_multicontact_control:BOOL=OFF
SUPERBUILD_multicontact_planning:BOOL=OFF
SUPERBUILD_network:BOOL=OFF
SUPERBUILD_nlopt:BOOL=OFF
SUPERBUILD_pilot_interface:BOOL=OFF
SUPERBUILD_psearch_control_interface:BOOL=OFF
SUPERBUILD_reactive_walking:BOOL=OFF
SUPERBUILD_rvl_perception:BOOL=OFF
SUPERBUILD_segway_v3:BOOL=OFF
SUPERBUILD_segway_v3_desktop:BOOL=OFF
SUPERBUILD_segway_v3_robot:BOOL=OFF
SUPERBUILD_segway_v3_simulator:BOOL=OFF
SUPERBUILD_sensors-description:BOOL=OFF
SUPERBUILD_sharedlibpp:BOOL=ON
SUPERBUILD_srdfdom_advr:BOOL=ON
SUPERBUILD_supervisor:BOOL=OFF
SUPERBUILD_trajectory_utils:BOOL=ON
SUPERBUILD_upper_body_control:BOOL=OFF
SUPERBUILD_walking:BOOL=OFF
SUPERBUILD_walkman_final_demo_field:BOOL=OFF
SUPERBUILD_walkman_fireman:BOOL=OFF
SUPERBUILD_wholebody_ik:BOOL=OFF
SUPERBUILD_yarp_ros_joint_state_publisher:BOOL=OFF
SUPERBUILD_yw_walking:BOOL=OFF
USE_DRC_CORE_PACKAGES:BOOL=OFF
USE_ONLY_OPEN_SOURCE:BOOL=ON
USE_WALKMAN_DEMOS_PROJECTS:BOOL=OFF
VIGIR:BOOL=OFF
WALKMAN_EU:BOOL=ON
WALKMAN_shared_DIR:PATH=WALKMAN_shared_DIR-NOTFOUND
WebGUIDistr_DIR:PATH=WebGUIDistr_DIR-NOTFOUND
XBotCoreModel_DIR:PATH=XBotCoreModel_DIR-NOTFOUND
XBotCore_DIR:PATH=XBotCore_DIR-NOTFOUND
XBotDummy_DIR:PATH=XBotDummy_DIR-NOTFOUND
XBotInterface_DIR:PATH=XBotInterface_DIR-NOTFOUND
YCM_DIR:PATH=YCM_DIR-NOTFOUND
YCM_EP_DEVEL_MODE_ADVR_shared:BOOL=OFF
YCM_EP_DEVEL_MODE_WALKMAN_shared:BOOL=OFF
YCM_EP_DEVEL_MODE_WebGUIDistr:BOOL=OFF
YCM_EP_DEVEL_MODE_iit-bigman-ros-pkg:BOOL=OFF
YCM_EP_DEVEL_MODE_iit-walkman-ros-pkg:BOOL=OFF
YCM_GIT_GITLAB_ADVRCLOUD_BASE_ADDRESS:STRING=ssh://git@gitlab.advrcloud.iit.it/
YNL_HYDRA_TOKYO_UNIVERSITY:BOOL=OFF
bprinter_DIR:PATH=bprinter_DIR-NOTFOUND
cpp_common_DIR:PATH=/opt/ros/kinetic/share/cpp_common/cmake
gazebo_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/gazebo
gazebo_magnetic_attach_plugin_DIR:PATH=gazebo_magnetic_attach_plugin_DIR-NOTFOUND
gazebo_proto_msgs_lib:FILEPATH=/usr/lib/x86_64-linux-gnu/libgazebo_msgs.so
ignition-math2_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/ignition-math2
iit-bigman-ros-pkg_DIR:PATH=iit-bigman-ros-pkg_DIR-NOTFOUND
iit-walkman-ros-pkg_DIR:PATH=iit-walkman-ros-pkg_DIR-NOTFOUND
lib:FILEPATH=/opt/ros/kinetic/lib/libxmlrpcpp.so
message_runtime_DIR:PATH=/opt/ros/kinetic/share/message_runtime/cmake
onelib:FILEPATH=/usr/lib/x86_64-linux-gnu/libignition-math2.so
rbdl_DIR:PATH=rbdl_DIR-NOTFOUND
rosconsole_DIR:PATH=/opt/ros/kinetic/share/rosconsole/cmake
roscpp_DIR:PATH=/opt/ros/kinetic/share/roscpp/cmake
roscpp_serialization_DIR:PATH=/opt/ros/kinetic/share/roscpp_serialization/cmake
roscpp_traits_DIR:PATH=/opt/ros/kinetic/share/roscpp_traits/cmake
rosgraph_msgs_DIR:PATH=/opt/ros/kinetic/share/rosgraph_msgs/cmake
rostime_DIR:PATH=/opt/ros/kinetic/share/rostime/cmake
sharedlibpp_DIR:PATH=sharedlibpp_DIR-NOTFOUND
srdfdom_advr_DIR:PATH=srdfdom_advr_DIR-NOTFOUND
std_msgs_DIR:PATH=/opt/ros/kinetic/share/std_msgs/cmake
trajectory_utils_DIR:PATH=trajectory_utils_DIR-NOTFOUND
xmlrpcpp_DIR:PATH=/opt/ros/kinetic/share/xmlrpcpp/cmake

**Comment by alessandrosettimi** [comment]
Putting "XBotCore" instead of "XCM" in the tag `<name>` of package.xml of XBotCore made things work. But I don't know if it is correct.

**Comment by liesrock** [comment]
@alaurenzi @pepperig what do you think?

**Comment by liesrock** [comment]
If you had one of the following repo enabled, you would have the XCM dependency (thanks @EnricoMingo ):

./BuildTestProjectGiuseppe.cmake:                DEPENDS XCM
./Buildqppvm.cmake:		DEPENDS XCM
./BuildWBObjectFollower.cmake:                DEPENDS XCM
./Buildkukatest.cmake:                DEPENDS XCM
./BuildEtherCAT_tools.cmake:		DEPENDS XCM
./Buildcentauro_bonn_review_plugins.cmake:		DEPENDS XCM
./BuildXBotGUI.cmake:                DEPENDS XCM
./Buildcentauro_upperbody_cheng.cmake:		DEPENDS XCM
./BuildDebrisRemovalTask.cmake:                DEPENDS XCM
./BuildParking_Steering.cmake:                DEPENDS XCM)
./BuildArtificialPotentialFields.cmake:		DEPENDS XCM
./BuildManipulationPlugin.cmake:		DEPENDS XCM
./BuildDoorOpening.cmake:                               DEPENDS XCM
./Buildmulti_contacts.cmake:find_or_build_package(XCM QUIET)
./Buildmulti_contacts.cmake:		DEPENDS XCM
./Buildwbc_tde.cmake:                DEPENDS XCM)
./BuildJuan_FBcontrollers.cmake:		DEPENDS XCM
./Buildlocomotion.cmake:                DEPENDS XCM
./BuildcentauroLocomotor.cmake:                DEPENDS XCM
./Buildchengxu_walking.cmake:		DEPENDS XCM
./Buildik_twolayer.cmake:                DEPENDS XCM)
./ProjectsTags.cmake:#set(XCM_TAG 9c9d211a4f39a304)
./BuildManipulationExample.cmake:		DEPENDS XCM
./BuildController.cmake:		DEPENDS XCM
./BuildPlayback_mat_with_DK.cmake:                DEPENDS XCM
./Buildyw_walking.cmake:		DEPENDS XCM
./BuildPholusWorkshop.cmake:# If XCM is not found, the superbuild will clone it and compile it.
./BuildPholusWorkshop.cmake:# XCM is the package which manages XBot RT (real-time) Plugins.
./BuildPholusWorkshop.cmake:		DEPENDS XCM

Solved with fec308b1f5d0e40031f232931a22ebbcbb1226f1

**Comment by alessandrosettimi** [comment]
I tried from a clean build after updating, worked. Thanks!
